src/main/scala/net/ripe/rpki/publicationserver/store/ObjectStore.scala

```diff
@@ -69,13 +69,30 @@ class ObjectStore extends Hashing with Logging {
     txn.getAll(OBJECT_ENTITY_NAME).foreach(e => e.delete())
   }
 
+  private def getBytes(e: Entity) = {
+    // This is to preserve back-compatibility with the base64-storing version.
+    // We do not store base64 (see fillEntity method), but for older objects
+    // it can be set, so we can use it instead of bytes.
+    val bytes = e.getBlob(BYTES_FIELD_NAME)
+    if (bytes != null) {
+      Bytes.fromStream(bytes)
+    } else {
+      val base64 = e.getProperty(BASE64_FIELD_NAME)
+      if (base64 != null) {
+        Bytes.fromBase64(Base64(base64.toString))
+      } else {
+        // This is just to prevent an NPE and it doesn't make a lot of sense
+        Bytes(Array())
```

Comment:
I think there is the invariant `base64 != null || bytes != null`? I would not mind an exception.

(With proper logging/alerting)

Reply:
Yes, this is the invariant, but I'm not sure I want an exception here, I would prefer to have behavior when broken object is pubslished is broken object is _somehow_ submitted/stored, instead of the whole server going bananas.

Reply:
At least warn log, or even error?

Reply:
Yeah, makes sense. Done.